app/composition/workflows/whatsapp-llm-failure-retry/workflow.ts

```
@@ -0,0 +1,24 @@
+export async function retryWhatsAppLlmFailureWorkflow(
+	params: WhatsAppLlmFailureRetryParams,
+): Promise<void> {
+	"use workflow";
+
+	const firstAttempt = await retryWhatsAppLlmResponseStep(params);
+	if (firstAttempt.status === "sent" || firstAttempt.status === "stale") {
+		return;
+	}
+
+	for (const delay of RETRY_DELAYS) {
+		await sleep(delay);
```

Biome (2.4.4)

[error] Avoid using await inside loops.

(lint/performance/noAwaitInLoops)